### PR TITLE
Add 'scalar' label_mode in image_dataset_from_directory and load images from directory root for Regression

### PR DESCRIPTION
**Pull request by eduardofv**
Feature proposal in #15440

# Describe the feature and the current behavior/state.

`image_dataset_from_directory` currently assumes that the dataset will be used in classification tasks, so the labels take some form of categorical values (integers, binary, etc). It also make strong assumptions on the directory structure in which the images are located: if a list of labels is passed, the images should be placed on subdirs named the same as the labels (even if the documentations says that the structure is ignored).

For regression purposes, I suggest a new `label_mode` called "scalar" in which labels can take a real value and are passed in labels as a list or tuple, and all images are located in the root of directory.

## Example:

You have a directory with images and a text file with their labels for regression which are floats (labels are ordered alphanumerically according to image name):

```
with open("labels.txt") as fin:
  labels = [float(item) for item in fin.readlines()]
data = image_dataset_from_directory("images_dir", labels=labels, label_mode='scalar')
```

# Will this change the current api? How?

The parameter label_mode will accept a new value scalar. This value will be passed to index_directory as a new parameter. Inside it, if label_mode is scalar then the directory indexing is not performed on the subdirs, only on root.

# Who will benefit from this feature?

Regression problems which uses images in a single directory

### CODE REVIEW

File: keras/preprocessing/image_dataset_test.py [hunk 1]
```diff
@@ -259,6 +261,19 @@ def test_image_dataset_from_directory_manual_labels(self):
     self.assertLen(batch, 2)
     self.assertAllClose(batch[1], [0, 1])
 
+  def test_image_dataset_from_directory_scalar(self):
+    if PIL is None:
+      return
+
+    directory = self._prepare_directory(num_classes=0, count=2)
+    dataset = image_dataset.image_dataset_from_directory(
```
Comment (fchollet) [inline keras/preprocessing/image_dataset_test.py:269]: I would recommend simply doing

```python
dataset = image_dataset.image_dataset_from_directory(directory, batch_size=8, image_size=(18, 18), shuffle=False, labels=None)
dataset = dataset.zip((dataset, [0.1, 0.2]))
```

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. I'm not sure a new API is needed here since you can just use `image_dataset_from_directory` to create an image-only dataset then zip it with your labels. It's less code and less API surface.

**Comment by eduardofv** [comment]
On a related matter... what about exposing `paths_and_labels_to_dataset` in `image_dataset.py`  like `tf.keras.preprocessing.image_dataset_from_paths(...)` so we can reuse this function when passing a list of image paths and corresponding labels? 

My use case: I have a CSV with image paths + categorical attributes + labels and would like to load the image dataset with labels. Does that sounds reasonable?
